spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/LocalOptimizer.scala

```diff
@@ -176,19 +183,30 @@ class LocalOptimizer[T: ClassTag] (
       if (gradientClippingParams.enableConstantClipping) {
         grad.clamp(gradientClippingParams.minValueClip, gradientClippingParams.maxValueClip)
       }
-      optimMethod.state.update("epoch", state.get("epoch"))
-      optimMethod.state.update("neval", state.get("neval"))
-      optimMethod.optimize(_ => (ev.fromType(loss), grad), weight)
+      currentOptimMethods.foreach { case (optimMethod, (weight, grad)) =>
+        optimMethod.state.update("epoch", state.get("epoch"))
+        optimMethod.state.update("neval", state.get("neval"))
+        optimMethod.optimize(_ => (ev.fromType(loss), grad), weight)
+      }
       val end = System.nanoTime()
       wallClockTime += end - start
       count += batch.size()
+      val hyperParameterLog = optimMethods.map{ case (moduleName, optimMethod) =>
```

Comment:
should we move this to a utility method? as in distriOptimizer this is also used